staging/src/k8s.io/kube-aggregator/pkg/apiserver/handler_proxy_test.go

```diff
@@ -261,7 +261,7 @@ func TestProxyHandler(t *testing.T) {
 					},
 				},
 			},
-			expectedStatusCode: http.StatusServiceUnavailable,
```

Comment:
I would still expect this to surface a 503 error to the caller

/hold

Reply:
This particular test case fails due to incorrect `APIService` configuration, that leads to `error trying to reach service: x509: certificate is valid for test-service.test-ns.svc, not ..svc`. It sounds like a configuration error because at this point we don't know if the server we are trying to reach is available.

Reply:
from a consistency perspective, if the periodic healthcheck for the APIService encountered this error, it would flip the APIService to unavailable, and this code would return a 503:

https://github.com/kubernetes/kubernetes/blob/e2d8f6c278011b2eabf5754c3274bc406731933c/staging/src/k8s.io/kube-aggregator/pkg/apiserver/handler_proxy.go#L125-L128

It's unclear to me why the same error encountered by the proxy synchronously would surface a different error to the requesting user

Reply:
to preserve the old behaviour we could change the default responder https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/kube-aggregator/pkg/apiserver/handler_proxy.go#L240 to return `http.StatusServiceUnavailable`, are you okay with that?

it seems that it's only used by the proxy - https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/kube-aggregator/pkg/apiserver/handler_proxy.go#L178

Reply:
done.

Now the default responder will return `StatusServiceUnavailable`, this aligns with the old behavior of the `Transport`. It looks like it safe to do so because the default responder wasn't wired before (https://github.com/kubernetes/kubernetes/pull/92217/files#diff-16a26bb0d342318c6eb3d03e6c5756e9R239)

Reply:
that seems better to me, thanks. will let @sttts ack before merge

/hold cancel